core/shared/src/main/scala/zio/stm/ZSTM.scala

```diff
@@ -161,6 +175,18 @@ final class ZSTM[-R, +E, +A] private[stm] (
   def commitEither: ZIO[R, E, A] =
     either.commit.absolve
 
+  /**
+   * Repeats this `STM` effect until its result satisfies the specified predicate.
+   */
+  final def doUntil(f: A => Boolean): ZSTM[R, E, A] =
+    flatMap(a => if (!f(a)) doUntil(f) else ZSTM.succeedNow(a))
```

Comment:
I'd invert the condition here, just to make it aligned with `doWhile`, but treat this as a nitpick.